src/modules/fancyzones/lib/FancyZonesData.cpp

```diff
@@ -60,6 +59,94 @@ namespace
 
         return tmpDirPath;
     }
+
+    bool DeleteRegistryKey(HKEY hKeyRoot, LPTSTR lpSubKey)
+    {
+        // First, see if we can delete the key without having
+        // to recurse.
+        auto result = RegDeleteKey(hKeyRoot, lpSubKey);
+
+        if (result == ERROR_SUCCESS)
+        {
+            return true;
+        }
+
+        HKEY hKey;
+        result = RegOpenKeyEx(hKeyRoot, lpSubKey, 0, KEY_READ, &hKey);
+
+        if (result != ERROR_SUCCESS)
+        {
+            // Error opening key
```

Comment:
If You were to check `RegOpenKeyEx(...) != ERROR_SUCCESS)` code would be self explanatory and this comment would not be needed.